eva-server/src/main/java/uk/ac/ebi/eva/server/ws/RegionWSServer.java

```diff
@@ -55,98 +64,70 @@ public RegionWSServer() {
+        VariantEntityRepository.RelationalOperator siftScoreOperator = VariantEntityRepository.RelationalOperator.NONE;
+        Double siftScoreValue = null;
+        if (siftScore != null && !siftScore.isEmpty()) {
+            siftScoreOperator = Utils.getRelationalOperatorFromRelation(siftScore);
+            siftScoreValue = Utils.getValueFromRelation(siftScore);
         }
 
-        if (histogram) {
-            if (regions.size() != 1) {
-                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                return setQueryResponse("Sorry, histogram functionality only works with a single region");
-            } else {
-                if (interval > 0) {
-                    queryOptions.put("interval", interval);
-                }
-                return setQueryResponse(
-                        variantMongoDbAdaptor.getVariantFrequencyByRegion(regions.get(0), queryOptions));
-
-            }
-        } else if (regionsSize <= 1000000) {
-            if (regions.isEmpty()) {
-                if (!queryOptions.containsKey("id") && !queryOptions.containsKey("gene")) {
-                    response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                    return setQueryResponse("Some positional filer is needed, like region, gene or id.");
-                } else {
-                    return setQueryResponse(variantMongoDbAdaptor.getAllVariants(queryOptions));
-                }
-            } else {
-                return setQueryResponse(variantMongoDbAdaptor.getAllVariantsByRegionList(regions, queryOptions));
-            }
+        List<Region> regions = Region.parseRegions(regionId);
+        List<VariantEntity> variantEntities;
+
+        PageRequest pageRequest = Utils.getPageRequest(queryOptions);
+
+        if (regions.size() > 1) {
+            variantEntities =
+                    variantEntityRepository.findByRegionsAndComplexFilters(regions, studies, consequenceType,
+                                                                           mafOperator, mafvalue, polyphenScoreOperator,
+                                                                           polyphenScoreValue, siftScoreOperator,
+                                                                           siftScoreValue, pageRequest);
+        } else if (regions.size() == 1) {
+            Region region = regions.get(0);
+            variantEntities =
+                    variantEntityRepository.findByRegionAndComplexFilters(region.getChromosome(), region.getStart(),
+                                                                          region.getEnd(), studies, consequenceType,
+                                                                          mafOperator, mafvalue, polyphenScoreOperator,
+                                                                          polyphenScoreValue, siftScoreOperator,
+                                                                          siftScoreValue, pageRequest);
```

Comment:
I don't think the complexity of maintaining two methods that can generate equivalent queries is worth the marginal performance gain, if there is no good reason to have two methods in the repository, one that checks for the chromosome and other for a list of regions we should unite them. That way we can do a query with only one region in the list and simplify the code.